Firestore/core/src/model/value_util.cc

```diff
@@ -496,39 +621,68 @@ bool Contains(google_firestore_v1_ArrayValue haystack,
   return false;
 }
 
-Message<google_firestore_v1_Value> NullValue() {
-  Message<google_firestore_v1_Value> null_value;
-  null_value->which_value_type = google_firestore_v1_Value_null_value_tag;
-  null_value->null_value = {};
+google_firestore_v1_Value NullValue() {
+  google_firestore_v1_Value null_value;
+  null_value.which_value_type = google_firestore_v1_Value_null_value_tag;
+  null_value.null_value = {};
   return null_value;
 }
 
 bool IsNullValue(const google_firestore_v1_Value& value) {
   return value.which_value_type == google_firestore_v1_Value_null_value_tag;
 }
 
-Message<google_firestore_v1_Value> MaxValue() {
-  Message<google_firestore_v1_Value> max_value;
-  max_value->which_value_type = google_firestore_v1_Value_map_value_tag;
-  max_value->map_value.fields_count = 1;
-  max_value->map_value.fields =
+google_firestore_v1_Value MinValue() {
+  google_firestore_v1_Value null_value;
+  null_value.which_value_type = google_firestore_v1_Value_null_value_tag;
+  null_value.null_value = {};
+  return null_value;
+}
+
+bool IsMinValue(const google_firestore_v1_Value& value) {
+  return IsNullValue(value);
+}
+
+google_firestore_v1_Value MaxValue() {
+  google_firestore_v1_Value max_value;
+  max_value.which_value_type = google_firestore_v1_Value_map_value_tag;
+  max_value.map_value.fields_count = 1;
+  max_value.map_value.fields =
       nanopb::MakeArray<google_firestore_v1_MapValue_FieldsEntry>(1);
-  max_value->map_value.fields[0].key = nanopb::MakeBytesArray("__type__");
-  max_value->map_value.fields[0].value.which_value_type =
+  max_value.map_value.fields[0].key = MaxValueFieldKey;
+  max_value.map_value.fields[0].value.which_value_type =
       google_firestore_v1_Value_string_value_tag;
-  max_value->map_value.fields[0].value.string_value =
-      nanopb::MakeBytesArray("__max__");
+  max_value.map_value.fields[0].value.string_value = MaxValueFieldValue;
   return max_value;
 }
 
 bool IsMaxValue(const google_firestore_v1_Value& value) {
-  return value == *MaxValue();
+  if (value.which_value_type != google_firestore_v1_Value_map_value_tag) {
+    return false;
+  }
+
+  if (value.map_value.fields_count != 1) {
+    return false;
+  }
+
+  if (nanopb::MakeStringView(value.map_value.fields->key) !=
```

Comment:
Instead of comparing the string value, you could also compare the pointer address first and only compare the string value if the pointer address is different. Here and below.